core/lib/spree/localized_number.rb

```diff
@@ -21,7 +21,14 @@ def self.parse(number)
       # then replace the locale-specific decimal separator with the standard separator if necessary
       number = number.gsub(separator, '.') unless separator == '.'
 
-      number.to_d
+      cast_to_d(number)
     end
+
+    def self.cast_to_d(value)
+      value.to_s.to_d
```

Comment:
I like this better. Also we can use it when we return the decimal instead of using a `cast_to_d` method. How would you feel about:

```ruby
    def self.parse
      ...

      # Handle empty string for ruby 2.4 compatibility
      BigDecimal.new(number.presence || 0)
    end
```